tools/crm_mon.c

```diff
@@ -3622,78 +4269,84 @@ print_xml_status(pe_working_set_t * data_set,
                 break;
         }
 
-        fprintf(stream, "        <node name=\"%s\" ", node->details->uname);
-        fprintf(stream, "id=\"%s\" ", node->details->id);
-        fprintf(stream, "online=\"%s\" ", node->details->online ? "true" : "false");
-        fprintf(stream, "standby=\"%s\" ", node->details->standby ? "true" : "false");
-        fprintf(stream, "standby_onfail=\"%s\" ", node->details->standby_onfail ? "true" : "false");
-        fprintf(stream, "maintenance=\"%s\" ", node->details->maintenance ? "true" : "false");
-        fprintf(stream, "pending=\"%s\" ", node->details->pending ? "true" : "false");
-        fprintf(stream, "unclean=\"%s\" ", node->details->unclean ? "true" : "false");
-        fprintf(stream, "shutdown=\"%s\" ", node->details->shutdown ? "true" : "false");
-        fprintf(stream, "expected_up=\"%s\" ", node->details->expected_up ? "true" : "false");
-        fprintf(stream, "is_dc=\"%s\" ", node->details->is_dc ? "true" : "false");
-        fprintf(stream, "resources_running=\"%d\" ", g_list_length(node->details->running_rsc));
-        fprintf(stream, "type=\"%s\" ", node_type);
-        if (pe__is_guest_node(node)) {
-            fprintf(stream, "id_as_resource=\"%s\" ", node->details->remote_rsc->container->id);
-        }
-
         if (group_by_node) {
             GListPtr lpc2 = NULL;
 
-            fprintf(stream, ">\n");
             for (lpc2 = node->details->running_rsc; lpc2 != NULL; lpc2 = lpc2->next) {
                 resource_t *rsc = (resource_t *) lpc2->data;
-
-                rsc->fns->print(rsc, "            ", print_opts | pe_print_rsconly, stream);
+                assert(false); // this case is not implemented (if there are running_rsc)
```

Comment:
The build fails on this assert (you probably need to include assert.c to get it here)

Reply:
Thanks, it compiles now.

Reply:
I'll also fix the other compilation errors in the next amend